contracts/HybridCustody.cdc

```diff
@@ -0,0 +1,529 @@
+    pub resource interface Account {
+        pub fun getAddress(): Address
+        pub fun isChildOf(_ addr: Address): Bool
+        pub fun getParentsAddresses(): [Address]
+        pub fun borrowAccount(): &AuthAccount?
+    }
+
+    // A ChildAccount shares the BorrowableAccount capability to itelf with ProxyAccount resources
+    pub resource interface BorrowableAccount {
+        access(contract) fun borrowAccount(): &AuthAccount
+    }
+
+    // Public methods anyone can call on a child account
+    pub resource interface ChildAccountPublic {
+        pub fun getParentsAddresses(): [Address]
+        pub fun getParentStatuses(): {Address: Bool}
+        pub fun getRedeemedStatus(addr: Address): Bool?
+
+        access(contract) fun setRedeemed(_ addr: Address)
+    }
+
+    // Accessible to the owner of the ChildAccount.
+    pub resource interface ChildAccountPrivate {
+        pub fun removeParent(parent: Address): Bool
+        pub fun publishToParent(
+            parentAddress: Address,
+            factory: Capability<&CapabilityFactory.Manager{CapabilityFactory.Getter}>,
+            filter: Capability<&{CapabilityFilter.Filter}>
+        )
+        pub fun giveOwnership(to: Address)
+        pub fun relinquishOwnership()
+    }
+
+    // Public methods exposed on a proxy account resource. ChildAccountPublic will share
+    // some methods here, but isn't necessarily the same
+    pub resource interface AccountPublic {
+        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
+        pub fun getAddress(): Address
+    }
+
+    // Methods only accessible to the designated parent of a ProxyAccount
+    pub resource interface AccountPrivate {
+        pub fun getCapability(path: CapabilityPath, type: Type): Capability?
+        pub fun getPublicCapability(path: PublicPath, type: Type): Capability?
+
+        access(contract) fun redeemedCallback(_ addr: Address)
+    }
+
+    // Entry point for a parent to borrow its child account and obtain capabilities or
+    // perform other actions on the child account
+    pub resource interface ManagerPrivate {
+        pub fun borrowAccount(id: UInt64): &{AccountPrivate, AccountPublic}?
+        pub fun removeChildByAddress(addr: Address)
+        pub fun removeChild(id: UInt64)
+        // TODO: Owned account methods
+    }
+
+    // Functions anyone can call on a manager to get information about an account such as
+    // What child accounts it has
+    pub resource interface ManagerPublic {
+        pub fun borrowAccountPublic(id: UInt64): &{AccountPublic}?
+        // pub fun getChildAddresses(): [Address]
+    }
+
+    /*
+    Manager
+    A resource for an account which fills the Parent role of the Child-Parent account
+    management Model. A Manager can redeem or remove child accounts, and obtain any capabilities
+    exposed by the child account to them.
+    */
+    pub resource Manager: ManagerPrivate, ManagerPublic {
```

Comment:
All resources should probably implement `MetadataViews.Resolver`, not sure about `ResolverCollection` since we have default implementations now. Happy to do both, though

Reply:
going to add a TODO here so we can address it in a later PR